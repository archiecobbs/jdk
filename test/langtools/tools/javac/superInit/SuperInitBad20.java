/*
 * @test /nodynamiccopyright/
 * @bug 8194743
 * @summary Verify inner class creation is disallowed before super()
 *
 * @compile/fail/ref=SuperInitBad20.out -XDrawDiagnostics SuperInitBad20.java
 */
class SuperInitBad20 {

    SuperInitBad20() {
        new Inner();    // illegal implicit reference to 'this'
        super();
    }

    class Inner {
    }
}
