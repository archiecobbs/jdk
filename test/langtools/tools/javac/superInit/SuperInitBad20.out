SuperInitBad20.java:11:9: compiler.err.cant.ref.before.ctor.called: this
1 error
